### PR TITLE
fixed food count to include staffers and guests

### PR DESCRIPTION
**Pull request by EliAndrewC**
We implemented a ``FOOD_COUNT`` property, but it only counted food purchased and not food which automatically goes to staffers and guests.

### CODE REVIEW

File: magstock/templates/regextra.html [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <script type="text/javascript">
     if (window.BADGE_TYPES) {
         BADGE_TYPES[0].description = 'Includes three day admission and campground fees';
```
Comment (EliAndrewC) [inline magstock/templates/regextra.html]: This is an unrelated fix for a JS error I noticed while I was testing this.  We should only update the description of the supporter badge type if that badge type actually exists.

File: magstock/templates/regextra.html [hunk 2]
```diff
@@ -28,7 +30,9 @@
         }
     };
     $(function () {
```
Comment (EliAndrewC) [inline magstock/templates/regextra.html]: Another unrelated fix; we should only move the food checkbox around if it actually exists :)

**Comment by EliAndrewC** [comment]
This is all tested now, so I'd appreciate it if someone (@binary1230 or @kitsuta) could look it over and merge it as soon as they get a chance; the MAGStock folks would like to have this merged tonight.

File: magstock/templates/regextra.html [hunk 3]
```diff
@@ -15,7 +17,7 @@
         });
     }
     var showOrHideAllergies = function () {
-        setVisible('#allergies', $.field('purchased_food').is(':checked'));
+        setVisible('#allergies', $.field('purchased_food').is(':checked') || {{ attendee.auto_food|jsonize }});
```
Comment (EliAndrewC) [inline magstock/templates/regextra.html]: This fix makes the allergy info always appear for guests/staffers.

File: magstock/__init__.py [hunk 4]
```diff
@@ -45,6 +52,10 @@ class Attendee:
     def food_cost(self):
         return c.FOOD_PRICE if self.purchased_food else 0
 
+    @property
+    def auto_food(self):
+        return self.badge_type in [c.STAFF_BADGE, c.GUEST_BADGE]
```
Comment (kitsuta) [inline magstock/__init__.py]: Hm, I notice that even though this property is here, it's defined a second time in the SessionMixin class. That could cause problems later if one is updated but not the other.

It's probably not a big deal since this is just the plugin for Magstock, though.

**Comment by kitsuta** [comment]
This looks good. Are you ready to merge?

File: magstock/__init__.py (hunk 4 above)
Comment (EliAndrewC) [inline magstock/__init__.py]: You're right, this is definitely a situation where the same logic is implemented in two places.

The reason for this is efficiency.  The only way to use this property in the ``SessionMixin`` class would be to actually load all of the results into memory so the property could be checked.  We want to be able to do a ``.count()`` on the query, which means we end up being forced into defining the same thing in two places for efficiency.

Hope that makes sense.

**Comment by EliAndrewC** [comment]
Yep, ready to merge - I tested it all locally on my dev box.

File: magstock/__init__.py (hunk 4 above)
Comment (kitsuta) [inline magstock/__init__.py]: Yeah, it makes sense that you can't use a property inside a query. Annoying, though. :P
